cuda_pathfinder/cuda/pathfinder/_dynamic_libs/load_nvidia_dynamic_lib.py

```diff
@@ -58,4 +61,11 @@ def load_nvidia_dynamic_library(libname: str) -> LoadedDL:
     Raises:
         RuntimeError: If the library cannot be found or loaded
     """
-    return _load_nvidia_dynamic_library_no_cache(libname)
+    pointer_size_bits = struct.calcsize("P") * 8
+    if pointer_size_bits != 64:
+        raise RuntimeError(
+            f"cuda.pathfinder.load_nvidia_dynamic_lib() requires 64-bit Python."
+            f" Currently running: {pointer_size_bits}-bit Python"
+            f" {sys.version_info.major}.{sys.version_info.minor}"
+        )
```

Comment:
Is this something we need to check or can we catch an error elsewhere? These types of input checks tend to add overhead that we'd like to avoid.

Reply:
This is catching the issue "just in time" because I figured that'll be the best way to ensure it happens when it's needed, and is sure not to be a false positive.

This function is decorated with `@functools.cache`, and the runtime for actually finding and loading the library is probably several orders of magnitude (millions?) more than the `struct.calcsize("P")` call.

We could move `struct.calcsize("P")` into another `@functools.cache` helper function, but considering added complexity vs runtime gain, I figure that's on the over-engineering side of the spectrum.

Please let me know if you think this is worth more effort.